[Tee Wen Rui] iP

currynia opened the pull request:
# Tissue
> Hi, I'm tissue [find me here](https://github.com/currynia/ip/)

Tissue is a bot. It allows you to
- chat with it
- store your items

All you need to do is 
1. Clone the repository
2. Run it on your machine 😄 

Features:
- [x] Save your task list
- [ ] Program for **you**
- [ ] ~Dance~

---
Here is the `main` function.
```java
public static void main(String[] args) {
        new Tissue("./data/", "tissue.csv").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

currynia committed:
Add initial skeletal version

currynia committed:
Add echo

currynia committed:
Implement Add and List

currynia committed:
Add mark as done

currynia committed:
Add Todos, Events and Deadlines

currynia committed:
Add Automated Text UI Testing

currynia committed:
Add error handling

currynia committed:
Add deleting tasks

currynia committed:
Add save

currynia committed:
Merge branch 'branch-Level-7'

currynia committed:
Change Deadline to store LocalDate object

currynia committed:
Merge branch 'branch-Level-8'

currynia committed:
Add more OOP

currynia committed:
Add error handling of invalid date input

currynia committed:
Add delete task from storage

currynia committed:
Package files

currynia committed:
Merge remote-tracking branch 'origin/add-gradle-support'

currynia committed:
Merge branch 'branch-Level-A'

currynia committed:
Add JUnit tests

currynia committed:
Merge branch 'branch-Level-A'

currynia committed:
Add JavaDoc comments

currynia committed:
Merge branch 'branch-A-JavaDoc'

currynia committed:
Match code to standards

currynia committed:
Merge branch 'branch-A-CodingStandard'

currynia committed:
Add find function

currynia committed:
Merge branch 'branch-Level-9'

currynia committed:
fix variable name

currynia committed:
Add checkstyle

currynia committed:
Merge branch 'branch-A-CheckStyle'

currynia committed:
Add JavaFx Gui

currynia committed:
Merge branch 'branch-Level-10'

choonzies commented on src/main/java/tissue/GuiUi.java:
```diff
@@ -0,0 +1,136 @@
+        this.storage = storage;
+    }
+
+    public String getGreetingMessage() {
```
Good naming of method as a verb!

choonzies commented on src/main/java/tissue/Storage.java:
```diff
@@ -0,0 +1,132 @@
+/**
+ * Storage class to handle all reading and writing of tasks to local disk.
+ */
+public class Storage {
```
Good noun naming for class names

tsulemornorfi replied:
Thanks :)

choonzies commented on src/main/java/tissue/Storage.java:
```diff
@@ -0,0 +1,132 @@
+     *
+     * @param task The task to save to the file.
+     */
+    public void save(Task task) {
```
Maybe `saveTask` will be better?

tsulemornorfi replied:
True

choonzies commented on src/main/java/tissue/Storage.java:
```diff
@@ -0,0 +1,132 @@
+     * @return The tasks stored in an array.
+     */
+    @SuppressWarnings("checkstyle:Indentation")
+    public ArrayList<Task> load() {
```
Maybe `loadTask` will be better?

tsulemornorfi replied:
True

choonzies commented on src/main/java/tissue/Storage.java:
```diff
@@ -0,0 +1,132 @@
+        return taskList;
+    }
+
+    private String parseTask(Task task) {
```
I like this `parseTask` naming! 😄

tsulemornorfi replied:
Thanks :)

choonzies commented on src/main/java/tissue/TaskList.java:
```diff
@@ -0,0 +1,81 @@
+     *
+     * @return Number of tasks.
+     */
+    public int size() {
```
`getTaskSize()` may be better?

choonzies commented on src/main/java/tissue/task/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+/**
+ * Wrapper for a deadline task.
+ */
+public class Deadline extends Task {
```
Perhaps DeadlineTask will be better?

choonzies commented on src/main/java/tissue/window/MainWindow.java:
```diff
@@ -0,0 +1,48 @@
+
+    private Tissue tissue;
+
+    private final Image userImage = new Image(this.getClass().getResourceAsStream("/images/user.png"));
+    private final Image tissueImage =
```
Good variable naming! However these variables are perhaps constants and require ALLCAPS?

choonzies reviewed (commented):
LGTM. Minor nits on naming standards

currynia committed:
Add assertion statements

Various assertion statements were added to prevent programmers errors.

As a step towards better code, the assertions were used as liberally
as possible.

currynia committed:
Refactor code

To improve code quality and readability, long functions have been
broken down into smaller separate ones.

currynia committed:
Merge pull request #1 from currynia/branch-A-Assertions

Add assertion statements

currynia committed:
Merge branch 'master' into branch-A-CodeQuality

currynia committed:
Merge pull request #2 from currynia/branch-A-CodeQuality

Refactor code

currynia committed:
Add help extension

currynia committed:
Merge branch 'branch-BCD-Extension'

currynia committed:
Fix headers

currynia committed:
Add title to window

currynia committed:
Add handling of errors

currynia committed:
Merge branch 'clean-up'

currynia committed:
Remove unused imports

currynia committed:
Update readme

currynia committed:
Add error handling and help for find

currynia committed:
Add bye handling
